stablehlo/dialect/StablehloOps.td

```diff
@@ -2663,21 +2663,23 @@ def StableHLO_DynamicReshapeOp: StableHLO_ShapedInterfaceOp<"dynamic_reshape",
       [HLO_SpeculatableIfAllInputsStaticAndShapeConstant, NoMemoryEffect]> {
   let summary = "DynamicReshape operation";
   let description = [{
-    This operation is a work in progress, so it is not yet included in
-    the StableHLO specification: https://github.com/openxla/stablehlo/issues/8.
+    Performs reshape of `operand` tensor to a `result` tensor.
 
-    Informally, this operation does the same thing as ReshapeOp except that the
-    result shape is specified dynamically via `output_shape`:
-    https://github.com/openxla/stablehlo/blob/main/docs/spec.md#reshape
+    See:
+    https://github.com/openxla/stablehlo/blob/main/docs/spec.md#dynamic_reshape
 
     Example:
     ```mlir
-    %0 = stablehlo.dynamic_reshape %arg0, %shape : (tensor<?xf32>, tensor<2xindex>) -> tensor<?x?xf32>
+    %0 = stablehlo.constant dense<[3, 2]> : tensor<2xi64>
+    %1 = stablehlo.dynamic_reshape %arg0, %0 : (tensor<2x3xi64>, tensor<2xi64>) -> tensor<3x2xi64>
     ```
   }];
 
-  let arguments = (ins HLO_Tensor:$operand, HLO_StaticDimensionTensor:$output_shape);
-  let results = (outs HLO_Tensor:$result);
+  let arguments = (ins
+    HLO_TensorOrPerAxisQuantizedTensor:$operand /*dynamic_reshape_i1*/,
```

Comment:
Do we want per-axis support? Doesn't that require knowing the value of the `?` dim?

cc @sdasgup3

Reply:
Or are there cases where the mixture of unbounded dynamism and per-axis are actually useful?

Reply:
>> Do we want per-axis support? Doesn't that require knowing the value of the ? dim?

Being a dynamic counterpart to the reshape op, I would imagine same quantization constraint on this op.  I have seens a few uses of stablehlo dynamic ops with quantzed element-type in stablehlo quantier https://github.com/tensorflow/tensorflow/blob/5f1eb4d0a5ad87fded028e4f433692349148c9e0/tensorflow/compiler/mlir/quantization/stablehlo/tests/passes/quantize_composite_functions.mlir#L172.




However, as you said, there are things that cannot be verified statically. For e.g. https://github.com/openxla/stablehlo/blob/main/docs/spec.md#constraints-66  C3 cannot be verified: The quantization dimension for per-axis case can be determined, but size for result might be unknown.